### PR TITLE
[KRYS-2548] change default gas limit logic for multi send

### PR DESCRIPTION
**Pull request by MrQKyber**

### CODE REVIEW

File: KyberNetwork/KyberNetwork/Screens/Tabbars/Invest/Coordinator/MultiSendCoordinator.swift [hunk 1]
```diff
@@ -462,6 +462,74 @@ extension MultiSendCoordinator: MultiSendApproveViewControllerDelegate {
...
+                output.append((item.0, KNGasConfiguration.approveTokenGasLimitDefault))
+              }
+              group.leave()
+            }
+          }
+          group.notify(queue: .global()) {
+            controller.coordinatorDidUpdateGasLimit(gas: output)
+          }
+        }
+      }
```
Comment (tungnguyen20) [inline KyberNetwork/KyberNetwork/Screens/Tabbars/Invest/Coordinator/MultiSendCoordinator.swift:532]: I think we can create a `getEstimateGasLimit` for abstracted transaction protocol so we can avoid duplicate logic like this.

Example
```
protocol GasLimitRequestable {
  func createGasLimitRequest() -> KNEstimateGasLimitRequest
}

extension EIP1559Transaction: GasLimitRequestable {}

extension SignTransaction: GasLimitRequestable {} 

extension KNGeneralProvider {
  func getEstimateGasLimit(transaction: GasLimitRequestable, completion: @escaping (Result<BigInt, AnyError>) -> Void) {
    let request = transaction.createGasLimitRequest()
    // send request here
  }
}
```

Comment (tungnguyen20) [reply KyberNetwork/KyberNetwork/Screens/Tabbars/Invest/Coordinator/MultiSendCoordinator.swift:532]: Also can apply it to other duplicated blocks

Comment (MrQKyber) [reply KyberNetwork/KyberNetwork/Screens/Tabbars/Invest/Coordinator/MultiSendCoordinator.swift:532]: In multi send estimate gas need to return special return value for multiple approve request so currently we can not refactor like this

**Review by toidv — APPROVED** [review]

**Comment by tungnguyen20** [comment]
It's okay to merge now. But other same blocks should be refactored in the future.

**Review by tungnguyen20 — APPROVED** [review]

**Pull request merged**